develop/tutorials/articles/192-item-selector/01-selecting-entities-using-the-item-selector-api.markdown

```diff
@@ -0,0 +1,356 @@
+    can be created by invoking with either `HttpServletRequest` or
+    `PortletRequest`. Use the following pattern to create your URL:
+    `RequestBackedPortletURLFactoryUtil.create(request)`.
+- `ItemSelectedEventName`: a unique JavaScript event name that is
+    triggered by the item selector when the element is selected.
+- `ItemSelectorCriterion`: the criterion (or array of criteria) that
+    specifies the element that should be selected.
+    
+Follow the pattern below to obtain the URL for the Item Selector:
+
+    List<ItemSelectorReturnType> desiredItemSelectorReturnTypes =  new
+    ArrayList<>();
+    
+    desiredItemSelectorReturnTypes.add(new *ItemSelectorReturnType());
+    
+    *ItemSelectorCriterion class *nameItemSelectorCriterion =   new
+    *ItemSelectorCriterion();
+    
+    *nameItemSelectorCriterion.setDesiredItemSelectorReturnTypes(
+    desiredItemSelectorReturnTypes);
+    
+    PortletURL itemSelectorURL = `_itemSelector.getItemSelectorURL`(
+    
+      requestBackedPortletURLFactory, "ItemSelectedEventName",
+    
+      *ItemSelectorCriterion);
+
+For example, the following configuration obtains the item selector URL for the
+`ImageItemSelectorCriterion`:
+
+    List<ItemSelectorReturnType> desiredItemSelectorReturnTypes =  new
+    ArrayList<>();
+    
+    desiredItemSelectorReturnTypes.add(new URLItemSelectorReturnType());
+    
+    ImageItemSelectorCriterion imageItemSelectorCriterion =   new
+    ImageItemSelectorCriterion();
+    
+    imageItemSelectorCriterion.setDesiredItemSelectorReturnTypes(
+    desiredItemSelectorReturnTypes);
+    
+    PortletURL itemSelectorURL = `_itemSelector.getItemSelectorURL`(
+    
+      requestBackedPortletURLFactory, "sampleTestSelectItem",
+    
+      imageItemSelectorCriterion);
+
+The return types are declared in an `ArrayList`. The `URLItemSelectorReturnType`
+is added to the list of desired return types, and the 
+`imageItemSelectorCriterion` is set as the kind of element to select. The 
+`ImageItemSelectorCriterion()` constructor is referenced so it can be used. The 
+return types are then set to the item selector criterion using the setter 
+`setDesiredItemSelectorReturnTypes()`. Finally, the item selector URL is created
+using the `getItemSelectorURL()` method, with the value of the JavaScript item 
+selected event name set to `sampleTestSelectItem`.
+
+The order of the item selector criteria is important because it decides the 
+order of the selection views. If you are using one item selector criteria, then 
+this is not an issue for you, but for multiple types of criteria listed in an 
+array this can be an issue. For example, if you pass the item selector criteria 
+in this order: `ImageItemSelectorCriterion`, `VideoItemSelectorCriterion` it 
+will display the selection views for images first and then for videos, but if 
+you reverse the item selector criteria it will display the video selection view 
+first and then the image selection view.
```

Comment:
Yeah, we mean the order of the tabs for the selection views. 

Let me put you an example that hopefully will help you to fully understand it:

1. We have the Documents and Media view that is displayed when the criterion ImageItemSelectorCriterion is invoked. The service ranking property of the view is 100 (property = {"item.selector.view.order:Integer=100"})
2. We have the Flicker view that is displayed when the criterion ImageItemSelectorCriterion is invoked. The service ranking of the view is 200 (property = {"item.selector.view.order:Integer=200"})
3. We have the URL view that is displayed when the criterion URLItemSelectorReturnType is invoked. The service ranking of the view is 150 (property = {"item.selector.view.order:Integer=150"})

Scenario 1: Then, if we invoke the item selector with the order: URLItemSelectorCriterion first and ImageItemSelectorCriterion second, the order of the tabs will be: URL view, Flicker view and Documents and Media. URL view will be first because it matches the first criterion (URLItemSelectorCriterion) while Flicker View and Documetns and Media view matches the second criterion (ImageItemSelectorCriterion). Then, in order to decide whether Flicker or Documents and Media goes first, we use the service.ranking to distinguish (because they both belong to the same criterion). The higher the service ranking is, the first it will appear.

Scenario 2: If we invoke the item selector with the order: ImageItemSelectorCriterion first and URLItemSelectorCriterion second, the order of the tabs will be: Flicker view,  Documents and Media view and URL view. 

Let me know if you have any additional question about this.